lib/beaker/dsl/helpers/puppet_helpers.rb

```diff
@@ -529,17 +538,15 @@ def stub_hosts_on(machine, ip_spec)
         #                            or a role (String or Symbol) that identifies one or more hosts.
         # @param ip_spec [Hash{String=>String}] a hash containing the host to ip
         #   mappings
-        # @example Stub puppetlabs.com on the master to 127.0.0.1
-        #   with_host_stubbed_on(master, 'forgeapi.puppetlabs.com' => '127.0.0.1') do
+        # @param alias_spec [Hash{String=>Array[String]] an hash containing the host to alias(es) mappings to apply
+        # @example Stub forgeapi.puppetlabs.com on the master to 127.0.0.1 with an alias forgeapi.example.com
+        #   with_host_stubbed_on(master, {'forgeapi.puppetlabs.com' => '127.0.0.1'}, {'forgeapi.puppetlabs.com' => ['forgeapi.example.com']}) do
         #     puppet( "module install puppetlabs-stdlib" )
         #   end
-        def with_host_stubbed_on(host, ip_spec, &block)
+        def with_host_stubbed_on(host, ip_spec, alias_spec={}, &block)
           begin
             block_on host do |host|
-              ip_spec.each_pair do |address, ip|
-                logger.notify("Stubbing address #{address} to IP #{ip} on machine #{host}")
-                on( host, puppet('resource', 'host', address, 'ensure=present', "ip=#{ip}") )
-              end
+              stub_hosts_on(host, ip_spec, aliases_spec)
             end
 
             block.call
```

Comment:
@johnduarte I have one question about teardown behavior here. Now that `with_host_stubbed_on` is calling `stub_hosts_on`, this makes me wonder about the teardown behavior.

When this method calls `stub_hosts_on`, a teardown will be added to the test, so this is an operation that happens over the scope of one entire test. However, `with_host_stubbed_on` is for passing a block to be executed, which assumes that these host stubbings will be cleaned up at the end of the block, and you can run code after the block that won't be affected by these stubs. Does the code below clean up the aliases correctly, so that block cleanup happens correctly, or do we have to wait for the test teardown procedure to fully remove these?